hecuba_core/src/TupleRowFactory.cpp

```diff
@@ -228,21 +266,176 @@ int TupleRowFactory::cass_to_c(const CassValue *lhs, void *data, int16_t col) co
+
+                    CassUuid cass_uuid = {*time_and_version, *clock_seq_and_node};
+                    cass_tuple_set_uuid(tuple, (size_t) n, cass_uuid);
+                    break;
+                }
+                case CASS_VALUE_TYPE_TIMEUUID: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_INET: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_DATE: {
+
+                    break;
+                }
+                case CASS_VALUE_TYPE_TIME: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_SMALL_INT: {
+                    int16_t *value = (int16_t *) row->get_element(n);
+                    cass_tuple_set_int16(tuple, (size_t) n, *value);
+                    break;
+                }
+                case CASS_VALUE_TYPE_TINY_INT: {
+                    int8_t *value = (int8_t *) row->get_element(n);
+                    cass_tuple_set_int8(tuple, (size_t) n, *value);
+                    break;
+                }
+                case CASS_VALUE_TYPE_LIST: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_MAP: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_SET: {
+                    //TODO
+                    break;
+                }
+                case CASS_VALUE_TYPE_TUPLE: {
+                    TupleRow **ptr = (TupleRow **) element_i;
+                    const TupleRow *inner_data = *ptr;
+                    auto TFACT = TupleRowFactory(metadata->at(n).pointer);
+                    unsigned long n_types = metadata->at(n).pointer->size();
+                    CassTuple *tuple = cass_tuple_new(n_types);
+                    TFACT.bind_tuple(statement, inner_data, bind_pos, tuple);
+                    cass_statement_bind_tuple(statement, bind_pos, tuple);
+                    break;
+                }
+                case CASS_VALUE_TYPE_UDT: {
+                    throw ModuleException("User defined types not supported");
+                }
+                case CASS_VALUE_TYPE_CUSTOM:
+                case CASS_VALUE_TYPE_UNKNOWN:
+                default:
+                    //TODO
+                    break;
+            }
+        } else {
+            //Element is a nullptr
+            CassError rc = cass_statement_bind_null(statement, bind_pos);
+            CHECK_CASS("TupleRowFactory: Cassandra binding query unsuccessful [Null value], column:" +
+                       localMeta->at(bind_pos).info[0]);
+        }
+        //else: key was nullptr and we don't bind anything
+    }
+}
+
+void TupleRowFactory::bind(CassStatement *statement, const TupleRow *row, u_int16_t offset, CassTuple *tuple,
+                           std::string type) const {
```

Comment:
We agreed on having a single method "bind", and the argument type should be either a "bool" is_tuple or an Enum .

I actually prefer  an enum with 2 cases:
- Regular_type
- Tuple_type

Reply:
Because comparing with text is ugly and prone to bugs, when there are other elegant ways